src/main/scala/cromwell/webservice/CromwellApiHandler.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.webservice
+
+import akka.actor.{ActorRef, Actor, Props}
+import akka.pattern.ask
+import akka.util.Timeout
+import cromwell.binding._
+import cromwell.engine._
+import cromwell.webservice.CromwellApiHandler.WorkflowStatus
+import cromwell.webservice.PerRequest.RequestComplete
+import spray.http.StatusCodes
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+object CromwellApiHandler {
+  sealed trait WorkflowManagerMessage
+  case class SubmitWorkflow(wdl: WdlSource, inputs: WorkflowInputs) extends WorkflowManagerMessage
+  case class WorkflowStatus(id: WorkflowId) extends WorkflowManagerMessage
+
+  def props(workflowManagerActorRef : ActorRef): Props = {
+    Props(new CromwellApiHandler(workflowManagerActorRef))
+  }
+}
+
+class CromwellApiHandler(workflowManagerActorRef : ActorRef) extends Actor {
+  import context.dispatcher
+
+  override def receive = {
+    case WorkflowStatus(id) =>
+      implicit val timeout = Timeout(2.seconds)
+      val future: Future[Option[WorkflowState]] = ask(workflowManagerActorRef, WorkflowManagerActor.WorkflowStatus(id)).mapTo[Option[WorkflowState]]
+
+      future.onComplete {
+        case Success(state:Option[WorkflowState]) =>
```

Comment:
You probably don't need the type on state in terms of the compiler and IMO it's a bit superfluous aesthetically but feel free to leave it here if it was for the latter reason and not the former. If you hadn't said to lay it on thick w/ the idiomatic oriented comments I'd not have even brought this up, better to have too many type annotations than too little